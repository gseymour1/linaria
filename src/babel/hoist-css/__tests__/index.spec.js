/* eslint-disable no-template-curly-in-string */
/* @flow */

import * as babel from 'babel-core';

function transpile(source) {
  return babel.transform(source, {
    presets: ['react'],
    plugins: [require.resolve('../index')],
    babelrc: false,
  });
}

describe('hoist-css babel plugin', () => {
  it('should hoist CSS from JSX from styles', () => {
    const { code } = transpile(`
      function MyComponent() {
        return (
          <article
            id="post"
            {...styles(css\`color: \${colors.red};\`, css\`font-size: 12px;\`)}
          />
        );
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should hoist named CSS from JSX from styles', () => {
    const { code } = transpile(`
      function MyComponent() {
        return (
          <article
            id="post"
            {...styles(css.named('test')\`color: \${colors.red};\`, css\`font-size: 12px;\`)}
          />
        );
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should not affect other template literals in styles', () => {
    const { code } = transpile(`
      function MyComponent() {
        <article
          id="post"
          {...styles(sass\`color: \${colors.red};\`, \`font-size: 12px;\`)}
        />
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should hoist CSS from JSX from className', () => {
    const { code } = transpile(`
      function MyComponent() {
        return (
          <article
            className={css\`color: \${colors.red};\`}
          />
        );
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should not affect other template literals in className', () => {
    const { code } = transpile(`
      function MyComponent() {
        <article
          className={sass\`color: \${colors.red};\`}
        />
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should hoist CSS from JSX from class', () => {
    const { code } = transpile(`
      function MyComponent() {
        return (
          <article
            class={css\`color: \${colors.red};\`}
          />
        );
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should not affect other template literals in class', () => {
    const { code } = transpile(`
      function MyComponent() {
        <article
          class={\`color: \${colors.red};\`}
        />
      }
    `);

    expect(code).toMatchSnapshot();
  });

  it('should noop if no styles or class present', () => {
    const { code } = transpile(`
      function MyComponent() {
        return <article id="post" />;
      }
    `);

    expect(code).toMatchSnapshot();
  });
});
