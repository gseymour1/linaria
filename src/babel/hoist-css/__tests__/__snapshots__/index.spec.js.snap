// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoist-css babel plugin should hoist CSS from JSX from class 1`] = `
"
function MyComponent() {
  return React.createElement(\\"article\\", {
    \\"class\\": _article
  });
}

var _article = css\`color: \${colors.red};\`;"
`;

exports[`hoist-css babel plugin should hoist CSS from JSX from className 1`] = `
"
function MyComponent() {
  return React.createElement(\\"article\\", {
    className: _article
  });
}

var _article = css\`color: \${colors.red};\`;"
`;

exports[`hoist-css babel plugin should hoist CSS from JSX from styles 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function MyComponent() {
  return React.createElement(\\"article\\", _extends({
    id: \\"post\\"
  }, styles(_article, _article2)));
}

var _article = css\`color: \${colors.red};\`;

var _article2 = css\`font-size: 12px;\`;"
`;

exports[`hoist-css babel plugin should noop if no styles or class present 1`] = `
"
function MyComponent() {
  return React.createElement(\\"article\\", { id: \\"post\\" });
}"
`;

exports[`hoist-css babel plugin should not affect other template literals in class 1`] = `
"
function MyComponent() {
  React.createElement(\\"article\\", {
    \\"class\\": \`color: \${colors.red};\`
  });
}"
`;

exports[`hoist-css babel plugin should not affect other template literals in className 1`] = `
"
function MyComponent() {
  React.createElement(\\"article\\", {
    className: sass\`color: \${colors.red};\`
  });
}"
`;

exports[`hoist-css babel plugin should not affect other template literals in styles 1`] = `
"var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function MyComponent() {
  React.createElement(\\"article\\", _extends({
    id: \\"post\\"
  }, styles(sass\`color: \${colors.red};\`, \`font-size: 12px;\`)));
}"
`;
